Documentation build fails on Python 3.13

In #1871 I attempted to updated the Python version used for building documentation from Python 3.10 to Python 3.13. That caused the doc build to fail (likely because of changes in the version of Sphinx or some other package rather than directly because of the Python version change).

Failing run: https://github.com/enthought/traits/actions/runs/20310589245/job/58340371045

Output:

```
Running Sphinx v9.0.4
loading translations [en]... done
making output directory... done
Converting `source_suffix = '.rst'` to `source_suffix = {'.rst': 'restructuredtext'}`.
[autosummary] generating autosummary for: changelog.rst, index.rst, traits_api_reference/base_trait_handler.rst, traits_api_reference/constants.rst, traits_api_reference/ctrait.rst, traits_api_reference/ctraits.rst, traits_api_reference/editor_factories.rst, traits_api_reference/has_traits.rst, traits_api_reference/index.rst, traits_api_reference/interface_checker.rst, ..., traits_user_manual/debugging.rst, traits_user_manual/deferring.rst, traits_user_manual/defining.rst, traits_user_manual/index.rst, traits_user_manual/installation.rst, traits_user_manual/internals.rst, traits_user_manual/intro.rst, traits_user_manual/listening.rst, traits_user_manual/notification.rst, traits_user_manual/testing.rst
loading intersphinx inventory 'pyface' from https://docs.enthought.com/pyface/objects.inv ...
loading intersphinx inventory 'python' from https://docs.python.org/3/objects.inv ...
loading intersphinx inventory 'traitsui' from https://docs.enthought.com/traitsui/objects.inv ...
building [mo]: targets for 0 po files that are out of date
writing output... 
building [html]: targets for 46 source files that are out of date
updating environment: [new config] 46 added, 0 changed, 0 removed
reading sources... [  2%] changelog
reading sources... [  4%] index
reading sources... [  7%] traits_api_reference/base_trait_handler
reading sources... [  9%] traits_api_reference/constants
reading sources... [ 11%] traits_api_reference/ctrait
reading sources... [ 13%] traits_api_reference/ctraits
reading sources... [ 15%] traits_api_reference/editor_factories
reading sources... [ 17%] traits_api_reference/has_traits
reading sources... [ 20%] traits_api_reference/index
reading sources... [ 22%] traits_api_reference/interface_checker
reading sources... [ 24%] traits_api_reference/trait_base
reading sources... [ 26%] traits_api_reference/trait_converters
reading sources... [ 28%] traits_api_reference/trait_dict_object
reading sources... [ 30%] traits_api_reference/trait_errors
reading sources... [ 33%] traits_api_reference/trait_factory
reading sources... [ 35%] traits_api_reference/trait_handler
reading sources... [ 37%] traits_api_reference/trait_handlers
reading sources... [ 39%] traits_api_reference/trait_list_object
reading sources... [ 41%] traits_api_reference/trait_notifiers
reading sources... [ 43%] traits_api_reference/trait_numeric
reading sources... [ 46%] traits_api_reference/trait_set_object
reading sources... [ 48%] traits_api_reference/trait_type
reading sources... [ 50%] traits_api_reference/trait_types
reading sources... [ 52%] traits_api_reference/traits
reading sources... [ 54%] traits_api_reference/traits.adaptation
reading sources... [ 57%] traits_api_reference/traits.etsconfig
reading sources... [ 59%] traits_api_reference/traits.observation
reading sources... [ 61%] traits_api_reference/traits.testing
reading sources... [ 63%] traits_api_reference/traits.util
reading sources... [ 65%] traits_api_reference/traits_listener
reading sources... [ 67%] traits_api_reference/version
reading sources... [ 70%] traits_tutorial/index
reading sources... [ 72%] traits_tutorial/introduction
reading sources... [ 74%] traits_user_manual/advanced
reading sources... [ 76%] traits_user_manual/appendix
reading sources... [ 78%] traits_user_manual/custom
reading sources... [ 80%] traits_user_manual/debugging
reading sources... [ 83%] traits_user_manual/deferring
reading sources... [ 85%] traits_user_manual/defining
reading sources... [ 87%] traits_user_manual/index
reading sources... [ 89%] traits_user_manual/installation
reading sources... [ 91%] traits_user_manual/internals
reading sources... [ 93%] traits_user_manual/intro
reading sources... [ 96%] traits_user_manual/listening
reading sources... [ 98%] traits_user_manual/notification
reading sources... [100%] traits_user_manual/testing
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
copying assets... 
copying downloadable files... [ 14%] ../../examples/introduction/0_introduction.py
copying downloadable files... [ 29%] ../../examples/introduction/1_validation.py
copying downloadable files... [ 43%] ../../examples/introduction/2_initialization.py
copying downloadable files... [ 57%] ../../examples/introduction/3_observation.py
copying downloadable files... [ 71%] ../../examples/introduction/4_properties.py
copying downloadable files... [ 86%] ../../examples/introduction/5_documentation.py
copying downloadable files... [100%] ../../examples/introduction/6_visualization.py
copying static files... 
Writing evaluated template result to /home/runner/work/traits/traits/docs/build/_static/language_data.js
Writing evaluated template result to /home/runner/work/traits/traits/docs/build/_static/basic.css
Writing evaluated template result to /home/runner/work/traits/traits/docs/build/_static/documentation_options.js
Writing evaluated template result to /home/runner/work/traits/traits/docs/build/_static/enthought.css
Writing evaluated template result to /home/runner/work/traits/traits/docs/build/_static/copybutton.js
copying static files: done
copying extra files... 
copying extra files: done
copying assets: done

Theme error!

Versions
========

* Platform:         linux; (Linux-6.11.0-1018-azure-x86_64-with-glibc2.39)
* Python version:   3.13.11 (CPython)
* Sphinx version:   9.0.4
* Docutils version: 0.22.3
* Jinja2 version:   3.1.6
* Pygments version: 2.19.2

Last Messages
=============

    Writing evaluated template result to /home/runner/work/traits/traits/docs/build/_static/enthought.css
    Writing evaluated template result to /home/runner/work/traits/traits/docs/build/_static/copybutton.js
    copying static files: done
    copying extra files...
    copying extra files: done
    copying assets: done

    writing output... [  2%]
    changelog

Loaded Extensions
=================

* sphinx.ext.mathjax (9.0.4)
* alabaster (1.0.0)
* sphinxcontrib.applehelp (2.0.0)
* sphinxcontrib.devhelp (2.0.0)
* sphinxcontrib.htmlhelp (2.1.0)
* sphinxcontrib.serializinghtml (2.0.0)
* sphinxcontrib.qthelp (2.0.0)
* sphinx.ext.autodoc (9.0.4)
* sphinx.ext.autosummary (9.0.4)
* sphinx.ext.coverage (9.0.4)
* sphinx.ext.githubpages (9.0.4)
* sphinx.ext.intersphinx (9.0.4)
* sphinx.ext.napoleon (9.0.4)
* sphinx.ext.viewcode (9.0.4)
* sphinx_copybutton (0.5.2)
* traits.util.trait_documenter (unknown version)

Traceback
=========

      File "/opt/hostedtoolcache/Python/3.13.11/x64/lib/python3.13/site-packages/sphinx/builders/html/__init__.py", line 1229, in handle_page
        raise ThemeError(msg) from exc
    sphinx.errors.ThemeError: An error happened in rendering the page changelog.
    Reason: TypeError("argument of type '_CascadingStyleSheet' is not iterable")


The full traceback has been saved in:
/tmp/sphinx-err-5b2s48rk.log

To report this error to the developers, please open an issue at <https://github.com/sphinx-doc/sphinx/issues/>. Thanks!
Please also report this if it was a user error, so that a better error message can be provided next time.
writing output... [  2%] changelog
Error: Process completed with exit code 2.
```

--- a/.github/workflows/test-documentation-build.yml
+++ b/.github/workflows/test-documentation-build.yml
@@ -11,7 +11,7 @@ jobs:
     - name: Set up Python
       uses: actions/setup-python@v6
       with:
-        python-version: '3.10'
+        python-version: '3.13'
     - name: Install dependencies and local packages
       run: python -m pip install .[docs]
     - name: Build HTML documentation with Sphinx

--- a/setup.py
+++ b/setup.py
@@ -296,7 +296,9 @@ setuptools.setup(
     },
     extras_require={
         "docs": [
-            "enthought-sphinx-theme",
+            # enthought-sphinx-theme 0.7.4 is the first release compatible
+            # with Sphinx 9 (xref: enthought/traits#1872).
+            "enthought-sphinx-theme>=0.7.4",
             "Sphinx>=2.1.0",
             "sphinx-copybutton",
         ],

--- a/traits/adaptation/adaptation_manager.py
+++ b/traits/adaptation/adaptation_manager.py
@@ -104,11 +104,11 @@ class AdaptationManager(HasTraits):
 
         Parameters
         ----------
-        adaptee : object
+        adaptee : any object
             The object that we want to adapt.
         to_protocol : type or interface
             The protocol that the want to adapt *adaptee* to.
-        default : object, optional
+        default : any object, optional
             Object to return if no adaptation is possible. If no default is
             provided, and adaptation fails, an ``AdaptationError`` is raised.
 

--- a/traits/base_trait_handler.py
+++ b/traits/base_trait_handler.py
@@ -64,11 +64,11 @@ class BaseTraitHandler(object):
 
         Parameters
         ----------
-        object : object
+        object : HasTraits
             The object whose attribute is being assigned.
         name : str
             The name of the attribute being assigned.
-        value : object
+        value : any object
             The proposed new value for the attribute.
         """
         raise TraitError(
@@ -94,11 +94,11 @@ class BaseTraitHandler(object):
 
         Parameters
         ----------
-        object : object
+        object : HasTraits
             The object whose attribute is being assigned.
         name : str
             The name of the attribute being assigned.
-        value :
+        value : any object
             The proposed new value for the attribute.
         """
         return self.info()

--- a/traits/ctraits.c
+++ b/traits/ctraits.c
@@ -1470,7 +1470,7 @@ PyDoc_STRVAR(
     "name : str\n"
     "    Name of the item trait for which an event is being fired. (The name\n"
     "    will usually end in '_items'.)\n"
-    "event_object : object\n"
+    "event_object : any object\n"
     "    Object of type ``TraitListEvent``, ``TraitDictEvent`` or ``TraitSetEvent``\n"
     "    describing the changes to the underlying collection trait value.\n"
     "event_trait : CTrait\n"
@@ -5152,7 +5152,7 @@ PyDoc_STRVAR(
     "-------\n"
     "default_value_type : int\n"
     "    An integer representing the kind of the default value\n"
-    "default_value : value\n"
+    "default_value : any object\n"
     "    A value or callable providing the default\n");
 
 PyDoc_STRVAR(
@@ -5165,7 +5165,7 @@ PyDoc_STRVAR(
     "----------\n"
     "default_value_type : int\n"
     "    An integer representing the kind of the default value\n"
-    "default_value : value\n"
+    "default_value : any object\n"
     "    A value or callable providing the default\n");
 
 PyDoc_STRVAR(
@@ -5184,7 +5184,7 @@ PyDoc_STRVAR(
     "\n"
     "Returns\n"
     "-------\n"
-    "default_value : value\n"
+    "default_value : any object\n"
     "    The default value for the given object and name.\n");
 
 PyDoc_STRVAR(
@@ -5241,7 +5241,7 @@ PyDoc_STRVAR(
     "    The HasTraits object that validation is being performed for.\n"
     "name : str\n"
     "    The name of the trait.\n"
-    "value : object\n"
+    "value : any object\n"
     "    The value to be validated.\n"
     "\n"
     "Returns\n"

--- a/traits/has_traits.py
+++ b/traits/has_traits.py
@@ -1550,7 +1550,7 @@ class HasTraits(CHasTraits, metaclass=MetaHasTraits):
 
         Parameters
         ----------
-        other : object
+        other : HasTraits
             The object whose trait attribute values should be copied.
         traits : list of strings
             A list of names of trait attributes to copy. If None or
@@ -1658,7 +1658,7 @@ class HasTraits(CHasTraits, metaclass=MetaHasTraits):
 
         Returns
         -------
-        new :
+        new : HasTraits
             The newly cloned object.
         """
         if memo is None:
@@ -1723,7 +1723,7 @@ class HasTraits(CHasTraits, metaclass=MetaHasTraits):
             **traitsui.view.kind_trait** trait for values and
             their meanings. If *kind* is unspecified or None, the **kind**
             attribute of the View object is used.
-        context : object or dictionary
+        context : any object or dictionary
             A single object or a dictionary of string/object pairs, whose trait
             attributes are to be edited. If not specified, the current object
             is used.
@@ -2054,7 +2054,7 @@ class HasTraits(CHasTraits, metaclass=MetaHasTraits):
 
             .. deprecated:: 6.2.0
 
-        context : object or dictionary
+        context : any object or dictionary
             A single object or a dictionary of string/object pairs, whose trait
             attributes are to be edited. If not specified, the current object
             is used
@@ -2665,7 +2665,7 @@ class HasTraits(CHasTraits, metaclass=MetaHasTraits):
         ----------
         name : str
             Name of the trait attribute on this object.
-        object : object
+        object : HasTraits
             The object with which to synchronize.
         alias : str
             Name of the trait attribute on *other*; if None or omitted, same

--- a/traits/observation/exception_handling.py
+++ b/traits/observation/exception_handling.py
@@ -39,7 +39,7 @@ class ObserverExceptionHandler:
 
         Parameters
         ----------
-        event : object
+        event : any object
             An event object emitted by the notification.
         """
         _logger.exception(
@@ -95,7 +95,7 @@ class ObserverExceptionHandlerStack:
 
         Parameters
         ----------
-        event : object
+        event : any object
             An event object emitted by the notification.
         """
         _, excp, _ = sys.exc_info()

--- a/traits/testing/optional_dependencies.py
+++ b/traits/testing/optional_dependencies.py
@@ -23,7 +23,7 @@ def optional_import(name):
 
     Returns
     -------
-    None or module
+    None or types.ModuleType
         None if the module is not available, and the module otherwise.
 
     """

--- a/traits/trait_handlers.py
+++ b/traits/trait_handlers.py
@@ -104,7 +104,7 @@ class TraitCoerceType(TraitHandler):
 
     Parameters
     ----------
-    aType : type or object
+    aType : type or any object
         Either a Python type or a Python value.  If this is an object, it is
         mapped to its corresponding type. For example, the string 'cat' is
         automatically mapped to ``str``.
@@ -256,7 +256,7 @@ class TraitInstance(TraitHandler):
 
     Parameters
     ----------
-    aClass : type, object or str
+    aClass : type, instance or str
         A Python type or a string that identifies the type, or an object.
         If this is an object, it is mapped to the class it is an instance of.
         If this is a str, it is either the name  of a class in the module

--- a/traits/trait_list_object.py
+++ b/traits/trait_list_object.py
@@ -394,7 +394,7 @@ class TraitList(list):
         ----------
         index : integer
             The position at which to insert.
-        object : object
+        object : any object
             The object to insert.
         """
 
@@ -418,7 +418,7 @@ class TraitList(list):
 
         Returns
         -------
-        item : object
+        item : any object
             The removed item.
 
         Raises
@@ -443,7 +443,7 @@ class TraitList(list):
 
         Parameters
         ----------
-        value : object
+        value : any object
             Value to be removed.
 
         Raises
@@ -754,7 +754,7 @@ class TraitListObject(TraitList):
         ----------
         index : integer
             The position at which to insert.
-        object : object
+        object : any object
             The object to insert.
         """
 
@@ -772,7 +772,7 @@ class TraitListObject(TraitList):
 
         Returns
         -------
-        item : object
+        item : any object
             The removed item.
 
         Raises
@@ -793,7 +793,7 @@ class TraitListObject(TraitList):
 
         Parameters
         ----------
-        value : object
+        value : any object
             Value to be removed.
 
         Raises

--- a/traits/trait_types.py
+++ b/traits/trait_types.py
@@ -152,7 +152,7 @@ class Any(TraitType):
 
     Parameters
     ----------
-    default_value : object, optional
+    default_value : any object, optional
         The default value for the trait. If this is an instance of either
         :class:`list` or :class:`dict` then a copy of the default value
         is made for each instance. Otherwise, the default is shared between
@@ -2575,7 +2575,7 @@ class List(TraitType):
 
     Attributes
     ----------
-    item_trait : trait
+    item_trait : CTrait
         The type of item that the list contains.
     minlen : integer
         The minimum length of a list that can be assigned to the trait.
@@ -2850,7 +2850,7 @@ class Set(TraitType):
 
     Attributes
     ----------
-    item_trait : a trait or value that can be converted to a trait
+    item_trait : CTrait
         The type of item that the set contains. If not specified, the set
         can contain items of any type.
     has_items : bool
@@ -2978,10 +2978,10 @@ class Dict(TraitType):
 
     Attributes
     ----------
-    key_trait : a trait
+    key_trait : CTrait
         The trait type for keys in the dictionary; if not specified, any
         values can be used as keys.
-    value_trait : a trait
+    value_trait : CTrait
         The trait type for values in the dictionary; if not specified, any
         values can be used as dictionary values.
     value_trait_handler : TraitHandler
@@ -3120,7 +3120,7 @@ class Map(TraitType):
             A dictionary whose keys are valid values for the trait attribute,
             and whose corresponding values are the values for the shadow
             trait attribute.
-        default_value : object, optional
+        default_value : any object, optional
             The default value for the trait. If given, this should be a key
             from the mapping. If not given, the first key from the mapping (in
             normal dictionary iteration order) will be used as the default.
@@ -3207,7 +3207,7 @@ class PrefixMap(TraitType):
         A dictionary whose keys are strings that are valid values for the
         trait attribute, and whose corresponding values are the values for
         the shadow trait attribute.
-    default_value : object, optional
+    default_value : any object, optional
         The default value for the trait. If given, this should be either a key
         from the mapping or a unique prefix of a key from the mapping. If not
         given, the first key from the mapping (in normal dictionary iteration

--- a/traits/traits.py
+++ b/traits/traits.py
@@ -550,7 +550,7 @@ def Property(
         **fget** and **fset**, and not look elsewhere on the class.
     handler : function
         A trait handler function for the trait.
-    trait : Trait or value
+    trait : Trait or any object
         A trait definition or a value that can be converted to a trait that
         constrains the values of the property trait.
     """

